data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -100,6 +101,16 @@ public KeyValueProcessor(final PluginMetrics pluginMetrics, final KeyValueProces
         }
 
         if (keyValueProcessorConfig.getIncludeKeys() != null) {
```

Comment:
This validation block should be simplified:

1. both includeKeys and excludeKeys in the config have default values so the null check is unnecessary
2. What if there is overlap between includeKeys and excludeKeys? Shall we also throw Exception? consider using Set.retainAll to calculate intersection between the includeKeys and excludeKeys.